packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart

```diff
@@ -19,7 +20,10 @@ class UrlLauncherPlugin extends UrlLauncherPlatform {
   /// Returns the newly created window.
   @visibleForTesting
   html.WindowBase openNewWindow(String url) {
-    return html.window.open(url, '');
+    // We need to open on _top in ios browsers in standalone mode.
+    // See https://github.com/flutter/flutter/issues/51461 for reference.
+    final target = navigator.standalone ? '_top' : '';
```

Comment:
Can you move this logic to the `launch` method, so we can pass it to the `openNewWindow` method?

I'd propose the openNewWindow to have this new signature:

```
html.WindowBase openNewWindow(String url, { String target = ''}) {
  return html.window.open(url, target);
}
```

And you'd call it like so:

```
openNewWindow(url, target: theTargetYouComputed)
```

Reply:
_(The reason for this is that some people want to be able to override the attribute manually; so that'd be a new (optional) parameter to the `launch` method. That might have to take into account the standalone computation as well!)_

Reply:
I totally agree on allow callers to specify the target but I think this should be handled as a separate thing with it's own issue/pr. What do you think?

Reply:
OK!